rust/projects/tools/project.md

```diff
@@ -114,19 +234,199 @@ this case the tests in `tests.rs` (`--test tests`). We can even narrow our testi
 down to a single test case within the `tests` test suite, like
 
 ```
-cargo test --test tests -- TODO
+cargo test --test tests -- cli_no_args
 ```
 
-And that's probably how you will be running the tests yourself as you work
+or, by matching multiple test cases, a few, like:
+
+
+```
+cargo test --test tests -- cli
+```
+
+_Try it now._
+
+That's probably how you will be running the tests yourself as you work
 through the project, otherwise you will be distracted by the many failing tests
-that you are not currently working on fixing.
+that you have not yet fixed.
 
 **Question D**: even if a given test suite doesn't contain any tests, why
 might we not want them to run? Besides issuing the above command, how could
 we permanently disable the three test suites we don't care about by
-editing the project manifest (`Cargo.toml`)? [Answers](answers.md#question-d).
+editing the project manifest (`Cargo.toml`)?
+[Answers](answers.md#question-d).
+
+
+## Part 2: Accept command line arguments
+
+The key / value stores throughout this course are all controlled through a
+command-line client. In this project the command-line client is very simple
+because the state of the key-value store is only stored in memory, not persisted
+to disk.
+
+In this part you will make the `cli_*` test cases pass. Notice that these test
+cases all call the `main` function from your `kvs` library. This is
+why we've put the "real" `main` function in `lib.rs` (the `kvs` library),
+instead of `main.rs` (the `kvs` CLI).
+
+**Question E**: This pattern of having the executable do nothing but call into
+the library's `main` function is often a reasonable thing to do, though it is
+not often used for production libraries. What are some downsides of placing a
+program's user interface into a library instead of directly into the executable?
+[Answers](answers.md#question-e).
+
+Recall how to run individual test cases from previous sections
+of 
+
+Again, the interface for the CLI is:
+
+- `kvs set [KEY] [VALUE]`
```

Comment:
I think KEY and VALUE are all required arguments. Is it better to change the square brackets to angle brackets `<>`?

Reply:
Yes, good catch.